storage/aptosdb/src/ledger_db/mod.rs

```diff
@@ -155,60 +156,95 @@ impl LedgerDb {
 
         let ledger_db_folder = db_root_path.as_ref().join(LEDGER_DB_FOLDER_NAME);
 
-        let event_db_raw = Arc::new(Self::open_rocksdb(
-            ledger_db_folder.join(EVENT_DB_NAME),
-            EVENT_DB_NAME,
-            &rocksdb_configs.ledger_db_config,
-            readonly,
-        )?);
-        let event_db = EventDb::new(event_db_raw.clone(), EventStore::new(event_db_raw));
-
-        let transaction_accumulator_db =
-            TransactionAccumulatorDb::new(Arc::new(Self::open_rocksdb(
-                ledger_db_folder.join(TRANSACTION_ACCUMULATOR_DB_NAME),
-                TRANSACTION_ACCUMULATOR_DB_NAME,
-                &rocksdb_configs.ledger_db_config,
-                readonly,
-            )?));
-
-        let transaction_auxiliary_data_db =
-            TransactionAuxiliaryDataDb::new(Arc::new(Self::open_rocksdb(
-                ledger_db_folder.join(TRANSACTION_AUXILIARY_DATA_DB_NAME),
-                TRANSACTION_AUXILIARY_DATA_DB_NAME,
-                &rocksdb_configs.ledger_db_config,
-                readonly,
-            )?));
-        let transaction_db = TransactionDb::new(Arc::new(Self::open_rocksdb(
-            ledger_db_folder.join(TRANSACTION_DB_NAME),
-            TRANSACTION_DB_NAME,
-            &rocksdb_configs.ledger_db_config,
-            readonly,
-        )?));
-
-        let transaction_info_db = TransactionInfoDb::new(Arc::new(Self::open_rocksdb(
-            ledger_db_folder.join(TRANSACTION_INFO_DB_NAME),
-            TRANSACTION_INFO_DB_NAME,
-            &rocksdb_configs.ledger_db_config,
-            readonly,
-        )?));
-
-        let write_set_db = WriteSetDb::new(Arc::new(Self::open_rocksdb(
-            ledger_db_folder.join(WRITE_SET_DB_NAME),
-            WRITE_SET_DB_NAME,
-            &rocksdb_configs.ledger_db_config,
-            readonly,
-        )?));
+        let mut event_db = None;
+        let mut transaction_accumulator_db = None;
+        let mut transaction_auxiliary_data_db = None;
+        let mut transaction_db = None;
+        let mut transaction_info_db = None;
+        let mut write_set_db = None;
+        THREAD_MANAGER.get_non_exe_cpu_pool().scope(|s| {
```

Comment:
seems no particular reason to use this pool?

Reply:
No. Just need some threads here.